planet_scope/ndwi/eob.js

```diff
@@ -0,0 +1,50 @@
+//VERSION=3
+//PlanetScope NDWI EO Browser
+
+function setup() {
+    return {
+        input: [{
+            bands: [
+                "green",
+                "nir",
+                "dataMask",
+                "clear"]
+        }],
+        output: [
+            { id: "default", bands: 4 },
+            { id: "index", bands: 1, sampleType: "FLOAT32" },
+            { id: "eobrowserStats", bands: 2, sampleType: "FLOAT32" },
+            { id: "dataMask", bands: 1 },
+        ]
+    }
+}
+
+function evaluatePixel(sample) {
+
+    let band1 = sample.green;
+    let band2 = sample.nir;
+    const denominator = band1 + band2;
+    let ndwi = denominator === 0 ? NaN : (band1 - band2) / denominator;
+
+    const clear = (sample.dataMask * sample.clear);
+
+    let id_default = colorBlend(ndwi,
+        [-1, -0.5, -0.2, 0, 0.2, 0.5, 1.0],
+        [
+            [1, 0, 1],
+            [1, 0.5, 0],
+            [1, 1, 0],
+            [0.2, 1, 0.5],
+            [0, 0, 1],
+            [0, 0, 0.3],
+            [0, 0, 0],
+        ]
+    );
+
+    return {
+        default: [...id_default, clear],
+        index: [ndwi],
+        eobrowserStats: [ndwi, sample.dataMask],
```

Comment:
```suggestion
        eobrowserStats: [ndwi, !sample.clear],
```

Reply:
should this be `!sample.clear` or `!clear`?  sample.clear is equal to the UDM mask, but clear is where I've used `&&` - with the intention of basically making it where the filter also removes statistics that might be calculated when only a fraction of the area is covered (like what if only 10% of a field is covered, but 100% of pixels are clear).

Reply:
I think clear, so I will add that there for now

Reply:
I see in the others that the suggestion was `!clear`, so will resolve

Reply:
actually thinking of it, `sample.clear` and `clear` should be basically the same. Since if `sample.dataMask` is 0, all band values are also either the set NoData value or if that doesn't exist, they are 0. So I think either is fine.